skosprovider/skos.py

```diff
@@ -581,8 +581,9 @@ def filter_labels_by_language(labels, language, broader=False):
     if language == 'any':
         return labels
     if broader:
-        language = tags.tag(language).language.format
-        return [l for l in labels if tags.tag(l.language).language.format == language]
+        if tags.tag(language).language:
```

Comment:
I'm not 100% satisfied with the solution. Since last release of the skosprovider we had some breaking changes. Requests that got results in an earlier release suddenly got 500 codes. This change would fix the 500 codes but we still wouldn't have a list of values as we did have before.

If no languages is given we should resolve to `language="und"`
https://github.com/OnroerendErfgoed/skosprovider/blob/24b79f4cd672c98c165bd4be50c7d41d6b3d1074/skosprovider/skos.py#L534-L535
I consider an empty string as language as an undifened language. I'm not sure what change in previous release altered the behaviour of a query like `https://thesaurus.onroerenderfgoed.be/conceptschemes/GEBEURTENISTYPES/c?ctype=concept&label=sy&language= `, but I would rather not break fixed requests coming from other applications

Reply:
> [give the same response as a call where the language key is not](https://thesaurus.onroerenderfgoed.be/conceptschemes/GEBEURTENISTYPES/c?ctype=concept&label=sy&language= should give the same response as a call where the language key is not present: https://thesaurus.onroerenderfgoed.be/conceptschemes/GEBEURTENISTYPES/c?ctype=concept&label=sy)

The behavior of `?language=` and no language was never the same. The language is set by the following code in pyramid_skosprovider https://github.com/OnroerendErfgoed/pyramid_skosprovider/blob/c7f24b2b0f8dc7cb3ca4b73b00f409c6f3f887f6/pyramid_skosprovider/utils.py#L19
When we have `?language=` language will become an empty string and when no language is present it will default to `self.request.locale_name` so if you want to change that behavior we would have to create an issue in pyramid_skosprovider. 

>This change would fix the 500 codes but we still wouldn't have a list of values as we did have before.

It will return a list like it used to do. `filter_labels_by_language` is only used in `find_best_label_for_type` which is only used in `label` https://github.com/OnroerendErfgoed/skosprovider/blob/79ed3f1497d0dabe6891d826287f23c41c94e3c6/skosprovider/skos.py#L481 
After it does not find a sortlabel, preflabel, altlabel it will default to `label(labels, 'any', sortLabel) if 'any' not in language else None` which will return the label with any language. 
![image](https://user-images.githubusercontent.com/45393594/149099681-3925254d-6717-40b2-84c7-dd62978f96d2.png)
The reason why `?language= ` now started to give errors, is probably because of https://github.com/OnroerendErfgoed/skosprovider/blob/79ed3f1497d0dabe6891d826287f23c41c94e3c6/skosprovider/skos.py#L534 the empty string is now first put into a list so the check `if not language:` will be False. Another fix would be to just replace that code with something like this `if not language or (len(language) == 1 and not language[0]:`. 

However while testing i discovered another bug. When you give a language` ?language=Bram` that does not exist it will still fail because `language = tags.tag(language).language.format` will still present an  error. The change I made will make it so that it will default to any language instead of returning 500. 

The most ideal case would be that the query param would be properly validated and any invalid language would give a` bad request`, but this would consist of a much larger change.

Reply:
ok, makes sense. 

I think I would prefer the alternative fix you mention, since it is a bit cleaner in my eyes and then we are guaranteed of minimal 1 language in the rest of the code. 
Based on your alternative fix: 1 line before the check on line 534 we could use a list comprehension to remove the empty values from the list: `language = [lang for lang in language if lang]`
So now the check will fail and the `und` value will be added.
I let you and @roefem make the final call. 

Testcase:
```
>>> l = ['']
>>> print(len(l))
1
>>> l = [li for li in l if li]
>>> print(len(l))
0
>>> if not l:
...     print('no lang')
... 
no lang

```